src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -406,11 +435,34 @@ public override object Clone()
             }
             else
             {
-                if (!trx.HasWitness && state.CorruptionPossible)
+                // Do not use rejection cache for witness transactions or
+                // witness-stripped transactions, as they can have been malleated.
+                // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
+                if (!trx.HasWitness && !state.CorruptionPossible)
                 {
+                    this.orphans.AddToRecentRejects(trxHash);
                 }
 
-                // TODO: Implement Processes whitelistforcerelay
+                // Always relay transactions received from whitelisted peers, even
+                // if they were already in the mempool or rejected from it due
+                // to policy, allowing the node to function as a gateway for
+                // nodes hidden behind it.
+                //
+                // Never relay transactions that we would assign a non-zero DoS
+                // score for, as we expect peers to do the same with us in that
+                // case.
+                if (this.isPeerWhitelistedForRelay)
+                {
+                    if (!state.IsInvalid)
```

Comment:
Can state be valid and we come to this point?
Ah this is in case `this.orphans.AlreadyHaveAsync` already has the trx?